### ISSUE TITLE
[Bug]: m in bold is not rendered correctly in Qt Creator

### ISSUE THREAD

**Issue by sonulohani** [MonoLisaFont/feedback#292]
### Edition

Plus

### What version are you using?

2.015

### What type of weights are you using?

Default (non-customized)

### Operating system

Ubuntu 24.04

### Program

Qt Creator

### Display resolution

1920x1080

### What happened?
![image](https://github.com/user-attachments/assets/44d3d38a-018e-42a1-9384-2ec4e6dc843f)

**Comment by bebraw** [comment]
Can you see the issue in other editors or at the website of MonoLisa for example? This information would help in reproducing the problem.

**Comment by sonulohani** [comment]
Nope, I can see this only in Qt creator

**Comment by nfejzic** [comment]
@sonulohani I had similar issues in other apps, in my case it was caused with FreeType font hinting. It improved when I disabled hinting and auto-hinting. Hope this helps.

**Comment by bebraw** [comment]
> @sonulohani I had similar issues in other apps, in my case it was caused with FreeType font hinting. It improved when I disabled hinting and auto-hinting. Hope this helps.

That's good info.

I found the following quote from https://freetype.org/autohinting/widths.html:

> These artefacts are rare, usually appear for a single point size which varies with the font, but still exist. A general principle of coherency is that it is better to be wrong by 10% all the time than completely correct in 95% cases, and greatly wrong in 5% cases.

Maybe "m" is hitting this due to autohinting by the operating system. I don't know if there's something specific that could be done at the font size but it's interesting to know it's the combination of Qt Creator and FreeType autohinting as that helps in debugging.

**Comment by nfejzic** [comment]
@bebraw I had this issue in two terminals, one of them was [wezterm], and another one is Ghostty (not public yet as far as I know), you can use those to test the font. 

As for the Ghostty, I could ask in discord for them to invite you to their github organisation so you can test it there as well. However, in Ghostty's case you'd have to test it on linux. Macos version uses CoreText rendering which does not have this issue (maybe another useful info 😄).

[wezterm]: https://wezfurlong.org/wezterm/

**Comment by bebraw** [comment]
@nfejzic Thanks, that's helpful. I have a feeling if we are able to fix one, it will fix the rest since it seems to be related to FreeType and how it renders. MacOS is known to render the best and there's some struggle under Windows with low resolution and their default hinting.

**Comment by sonulohani** [comment]
> @sonulohani I had similar issues in other apps, in my case it was caused with FreeType font hinting. It improved when I disabled hinting and auto-hinting. Hope this helps.

Right, Setting Hinting to None fixed the issue. If I set to Slight, then also 'm' renders fine.

**Comment by sonulohani** [comment]
I am able to reproduce this again in ghostty terminal, Can you fix it @bebraw ?

**Comment by bebraw** [comment]
@sonulohani I cannot promise a fix since it's hinting related but let's keep the issue open so we won't forget. I wonder if ghostty has some hinting related options to tweak.

**Comment by mfocko** [comment]
https://ghostty.org/docs/config/reference#freetype-load-flags

**Comment by bebraw** [comment]
@mfocko Thanks! I think the variant disabling hinting is worth a shot.

**Comment by nfejzic** [comment]
> https://ghostty.org/docs/config/reference#freetype-load-flags

Fun fact: this feature was implemented by me because I was annoyed with some letters rendering weird. So yeah, using the `no-hinting` flag is the way to go here 😄
